src/guest_memory.rs

```diff
@@ -494,15 +494,15 @@ impl<T: GuestMemory> Bytes<GuestAddress> for T {
                 // It is safe to read from volatile memory. Accessing the guest
                 // memory as a slice is OK because nothing assumes another thread
                 // won't change what is loaded.
-                let bytes_written = dst.write(&src[start..end]).map_err(Error::IOError)?;
-                Ok(bytes_written)
+                dst.write_all(&src[start..end]).map_err(Error::IOError)?;
```

Comment:
I am not sure about this, but isn't the implementation of write different from the implementation of write_all? In some cases write_all is blocking while write is not blocking. Do we want this call to block until all bytes are written?

Reply:
nit: Unrelated to this line, but can you please re-work your commit message so you don't have to have the summary on two lines? "guest_memory: avoid multiple reads from the same range" should suffice as a summary.

Related to commit: 72b073aceadc72b6de0d2693c0235b3adc15e313

Reply:
I undid this change. However, write_all unfortunately must remain in the other branch, because reading could have side effects.